test/persistence-hooks.suite.js

```diff
@@ -2946,6 +2946,252 @@ module.exports = function(dataSource, should, connectorCapabilities) {
 
     });
 
+    describe('PersistedModel.upsertWithWhere', function() {
+      var createdId = null;
+      var createdExtra = null;
+      var createdName = null;
+
+      it('triggers hooks in the correct order on create', function(done) {
+        monitorHookExecution();
+        TestModel.upsertWithWhere({ where: { extra: 'e0101' }},
+          { id: 'not-found', name: 'not found' },
+          function(err, record, created) {
+            if (err) return done(err);
+            createdId = record.id;
+            createdExtra = record.extra;
+            createdName = record.name;
+            hookMonitor.names.should.eql([
+              'access',
+              'before save',
+              'persist',
+              'loaded',
+              'after save',
+            ]);
+            done();
+          });
+      });
+
+      it('triggers hooks in the correct order on update', function(done) {
+        monitorHookExecution();
+        TestModel.upsertWithWhere({ where: { extra: 'e0101' }},
+          { name: 'new name' },
+          function(err, record, created) {
+            if (err) return done(err);
+            hookMonitor.names.should.eql([
+              'access',
+              'before save',
+              'persist',
+              'loaded',
+              'after save',
+            ]);
+            done();
+          });
+      });
```

Comment:
>I observed , it never validates the else part. It takes [] and [ values].

I'm not sure what you mean by this; [here](https://github.com/strongloop/loopback-datasource-juggler/blob/a22ff61af2d7c3edc97705905b4ea6c5566a00da/test/persistence-hooks.suite.js#L609-L633) is an example of test which we see different order of hooks for atomic implementtaion Vs. non-atomic implementation.

However in your case I believe your problem is that you are missing `var opts = { notify: false };` before calling non-atomic implementation [here](https://github.com/strongloop/loopback-datasource-juggler/pull/1001/files#r71822963); in other words by adding `var opts = { notify: false };`, I guess you would get the same order of hooks for atomic implementation Vs. non-atomic implementation.